### PR TITLE
feat(fileDND): #MAG-408 add file dnd from explorer

### PR DESCRIPTION
**Pull request by ThomasSevagen**
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/components/board-view/BoardView.tsx [hunk 1]
```diff
@@ -38,9 +39,14 @@ export const BoardView: FC = () => {
     hasEditRights,
     displayModals,
     toggleBoardModals,
+    setIsFileDragging,
+    isFileDragging,
   } = useBoard();
   const headerHeight = useHeaderHeight();
+  const dragCounter = useRef(0);
```
Comment (niko-amosse) [inline frontend/src/components/board-view/BoardView.tsx:46]: pourquoi pas un state ?????

File: frontend/src/components/file-drop-zone/FileDropZone.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  const {
+    board: { title },
+  } = useBoard();
+  const { t } = useTranslation("magneto");
+
+  const validateFile = (dataTransfer: DataTransfer | null): boolean => {
+    if (!dataTransfer?.types) return false;
+    const isFile =
+      dataTransfer.types.includes("Files") &&
+      !dataTransfer.types.includes("application/x-moz-file");
```
Comment (niko-amosse) [inline frontend/src/components/file-drop-zone/FileDropZone.tsx:35]: const ?

File: frontend/src/components/file-drop-zone/style.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    zIndex: "1000",
+  }),
+);
+
+export const InnerBox = styled(Box)<{ isDragging: boolean }>(
+  ({ isDragging }) => ({
+    position: "relative",
+    width: "100%",
+    height: "100%",
+    mt: isDragging ? 2 : 0,
```
Comment (niko-amosse) [inline frontend/src/components/file-drop-zone/style.ts:22]: à revoir

File: frontend/src/providers/MediaLibraryProvider/index.tsx [hunk 4]
```diff
@@ -105,16 +109,31 @@ export const MediaLibraryProvider: FC<MediaLibraryProviderProps> = ({
...
+      });
+      return setIsCreateMagnetOpen(true);
     }
   };
 
   useEffect(() => {
     updateLibraryMedia();
-  }, [libraryMedia]);
+  }, [libraryMedia, workspaceElement]);
+  console.log(media, getMediaLibraryType(workspaceElement?.name));
```
Comment (niko-amosse) [inline frontend/src/providers/MediaLibraryProvider/index.tsx:131]: loooooog

File: frontend/src/providers/MediaLibraryProvider/utils.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  if (EXTENSION_FORMAT.VIDEO.includes(fileExtension)) {
+    return "video";
+  }
+  if (EXTENSION_FORMAT.IMAGE.includes(fileExtension)) {
+    return "image";
+  }
+  if (EXTENSION_FORMAT.AUDIO.includes(fileExtension)) {
+    return "audio";
+  }
+  return "attachment";
```
Comment (niko-amosse) [inline frontend/src/providers/MediaLibraryProvider/utils.ts:23]: genre 0 const là ???

**Review by niko-amosse — APPROVED** [review]

**Review by FlorentMr — COMMENTED** [review]
Fix les quelques ptites consts manquantes

**Review by Mattox0 — APPROVED** [review]

**Review by alicedraillard — APPROVED** [review]

**Pull request merged**